test/unit/api/multi-profile.test.ts

```diff
@@ -0,0 +1,110 @@
+import { describe, expect, it } from "bun:test";
+import { USCoreBodyWeightProfileProfile } from "../../../examples/typescript-us-core/fhir-types/hl7-fhir-us-core/profiles/UscoreBodyWeightProfile";
+import { USCoreBodyHeightProfileProfile } from "../../../examples/typescript-us-core/fhir-types/hl7-fhir-us-core/profiles/UscoreBodyHeightProfile";
+import type { Observation } from "../../../examples/typescript-us-core/fhir-types/hl7-fhir-r4-core/Observation";
+
+const createObservation = (): Observation => ({ resourceType: "Observation", status: "final", code: {} });
+
+describe("Multi-Profile Pattern", () => {
+	describe("Body Weight Profile", () => {
+		it("creates valid body weight observation", () => {
+			const profile = new USCoreBodyWeightProfileProfile(createObservation());
+			profile.setVscat({});
+
+			const resource = profile.toResource();
+			expect(resource.resourceType).toBe("Observation");
+			expect(resource.category).toBeDefined();
+			expect(resource.category?.length).toBeGreaterThan(0);
+		});
+
+		it("has getters for slices", () => {
+			const profile = new USCoreBodyWeightProfileProfile(createObservation());
+			profile.setVscat({});
+
+			const vscat = profile.getVscat();
+			expect(vscat).toBeDefined();
+		});
+	});
+
+	describe("Body Height Profile", () => {
+		it("creates valid body height observation", () => {
+			const profile = new USCoreBodyHeightProfileProfile(createObservation());
+			profile.setVscat({});
+
+			const resource = profile.toResource();
+			expect(resource.resourceType).toBe("Observation");
+			expect(resource.category).toBeDefined();
+			expect(resource.category?.length).toBeGreaterThan(0);
+		});
+
+		it("has getters for slices", () => {
+			const profile = new USCoreBodyHeightProfileProfile(createObservation());
+			profile.setVscat({});
+
+			const vscat = profile.getVscat();
+			expect(vscat).toBeDefined();
+		});
+	});
+
+	describe("Profile composition patterns", () => {
+		it("wraps existing resource with profile", () => {
+			// Create a base observation
+			const baseObservation: Observation = {
+				resourceType: "Observation",
+				status: "final",
+				code: { coding: [{ system: "http://loinc.org", code: "29463-7", display: "Body Weight" }] },
+			};
+
+			// Wrap with body weight profile
+			const weightProfile = new USCoreBodyWeightProfileProfile(baseObservation);
+			weightProfile.setVscat({});
+
+			const resource = weightProfile.toResource();
+			expect(resource.status).toBe("final");
+			expect(resource.code?.coding?.[0]?.code).toBe("29463-7");
+			expect(resource.category).toBeDefined();
+		});
+
+		it("creates separate weight and height observations", () => {
```

Comment:
Looks like we can create them on top of one observation and look at what will happen.

Maybe with a check of the original resource shape at the end.